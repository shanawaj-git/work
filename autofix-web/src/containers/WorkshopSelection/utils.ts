import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
