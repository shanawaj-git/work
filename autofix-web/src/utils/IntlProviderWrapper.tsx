import React, { useMemo } from "react";
import { getTranslation, TranslationsType } from "../translations";
import { IntlProvider } from "react-intl";
import { flatData, I18NLocale } from ".";

export const withWrapper = (component: JSX.Element, pageProps = { en: {} }) => (
  <IntlProviderWrapper locale="en" pageProps={pageProps}>
    {component}
  </IntlProviderWrapper>
);

export const IntlProviderWrapper = ({
  children,
  pageProps,
  locale,
}: {
  children: JSX.Element;
  pageProps: TranslationsType;
  locale: I18NLocale;
}) => {
  const messages = useMemo(() => {
    return {
      ...getTranslation(locale),
      ...flatData(pageProps[locale] || {}),
    };
  }, [locale, pageProps]);

  return (
    <IntlProvider locale={`${locale}`} messages={messages}>
      {children}
    </IntlProvider>
  );
};
