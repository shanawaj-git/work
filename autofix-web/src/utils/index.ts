import { itemType } from "@/containers/RecommendedMaintenance";
import {
  AssetType,
  fetchCommonData,
  FooterType,
  getAssets,
  getAssetsData,
  getLabelData,
  getLabels,
  HomeActionType,
  LabelType,
  ServiceType,
} from "@/services/common";
import { TestimonialType } from "@/services/home/testimonial";

export const utils = "UTILS";
const GRAPHQL_CLIENT_READ_TIMEOUT = 5000;

interface getCommonPageContentParams {
  locale: I18NLocale;
  pageName: string;
}

export interface HomeContentType {
  labels?: LabelType[];
  assets?: AssetType[];
  services?: ServiceType[];
  actions?: HomeActionType[];
  testimonials?: TestimonialType[];
  footer?: FooterType;
}

export type I18NLocale = "en" | "ar";

/**
 * Will Return a Data object with Labels and Assets Regarding the page from Strapi
 * @param  locale , pageName
 * @returns Object with Labels and Assets
 */

export const getCommonPageContent = async ({
  locale,
  pageName,
}: getCommonPageContentParams): Promise<HomeContentType> => {
  let data: HomeContentType = {};
  return new Promise(async (resolve, reject) => {
    try {
      const commonData = await (
        await fetchCommonData(locale as I18NLocale, pageName)
      ).data;
      const assets = getAssets(commonData);
      const assetsData = getAssetsData(assets);
      const labels = getLabelData(getLabels(commonData));
      data = { assets: assetsData, labels };
      resolve(data);
    } catch (error) {
      resolve(data);
    }
  });
};

interface rejectEarlyType<T> {
  fn: () => Promise<T>;
  def: T;
  timeout?: number;
}

export const rejectEarly = <T>({
  fn,
  def,
  timeout = GRAPHQL_CLIENT_READ_TIMEOUT,
}: rejectEarlyType<T>): Promise<T> => {
  return new Promise(async (resolve, _reject) => {
    setTimeout(() => {
      resolve(def);
    }, timeout);
    try {
      const out: T = await fn();
      resolve(out);
    } catch (error) {
      resolve(def);
    }
  });
};

export const validateEmail = (email: string) =>
  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);

export const flatData = (obj: { [key: string]: {} }) =>
  Object.values(obj).reduce((acc: {}, curr: {}) => {
    return { ...acc, ...curr };
  }, {});

export const buildImageUrl = (path: string): string =>
  process.env.NEXT_PUBLIC_STRAPI_SERVICE?.concat(path) || "";

export const groupBy = <T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => string
) =>
  array.reduce((acc, value, index, array) => {
    (acc[predicate(value, index, array)] ||= []).push(value);
    return acc;
  }, {} as { [key: string]: T[] });

export const includesCaseInsensitive = (
  string: string,
  substring: string
): Boolean => {
  return string.toLowerCase().includes(substring.toLowerCase());
};
export const getTotalForMax = (data: itemType[]) => {
  return data.reduce((acc: number, curr: { cost: string }) => {
    let value = acc;
    value += +parseFloat(
      (curr.cost.match(/[0-9]+/gm) || ["0"]).splice(-1)[0]
    ).toFixed(2);
    return value;
  }, 0);
};

export const getCurrency = (data: itemType[]) => {
  return (data[0].cost.match(/[A-Z]+/gm) || ["USD"]).splice(-1)[0];
};

export const getCurrentLocation = () => {
  return new Promise((resolve: (e: GeolocationCoordinates) => void, reject) => {
    if (!navigator) throw new Error("Navigator doesn't exist");
    if (!navigator.geolocation) throw new Error("Geolocation doesn't exist");
    navigator.geolocation.getCurrentPosition(
      (position: GeolocationPosition) => {
        return resolve(position.coords);
      },
      reject,
      {
        maximumAge: 5000,
        enableHighAccuracy: true,
      }
    );
  });
};

interface getCurrentLiveLocationType {
  callBack: (e: GeolocationPosition) => void;
  onError: (e: GeolocationPositionError | Error) => void;
}

export const getCurrentLiveLocation = ({
  callBack,
  onError,
}: getCurrentLiveLocationType) => {
  if (!navigator) throw new Error("Navigator doesn't exist ");
  if (!navigator.geolocation) throw new Error("Geolocation doesn't exist ");
  navigator.geolocation.watchPosition((position: GeolocationPosition) => {
    callBack(position);
  }, onError);
};
