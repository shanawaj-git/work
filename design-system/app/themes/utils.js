import { getTheme } from './getTheme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
  focus: `0px 0px 10px ${getTheme().theme.color.d100}`,
};

const boxShadowSelector = level => BOX_SHADOW_MAPPING[level];
const colorSelector = color => getTheme().theme.color[color.toLowerCase()];
export { boxShadowSelector, colorSelector };
