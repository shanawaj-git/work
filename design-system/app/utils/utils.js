import { v4 } from 'uuid';
import _debounce from 'lodash/debounce';
/**
 * Helper: Convert Array of coordinates to an object of cordinates
 * @param {*} cordinates : Array [lat,lng]
 * @returns
 */
export const transformedCordinates = cordinates => {
  const [lat, lng] = cordinates;
  return {
    lat,
    lng,
  };
};

export const readableCordinates = ({ lat, lng }) => ({
  lat: lat(),
  lng: lng(),
});

export const uuid = () => v4();

const MAP_ADDRESS_COMPONENT_TYPE = {
  LOCALITY: 'locality',
  COUNTRY: 'country',
};

export const getCityAndCountry = address => {
  let city;
  let country;
  // eslint-disable-next-line no-unused-expressions
  address?.forEach(component => {
    const { types } = component;
    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.LOCALITY) > -1) {
      city = component.long_name;
    }

    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.COUNTRY) > -1) {
      country = component.long_name;
    }
  });
  return {
    city,
    country,
  };
};

export const debounce = (func, wait) => _debounce(func, wait);

export const canBeRenderedInTypographyComponent = obj =>
  typeof obj === 'string' ||
  obj?.type?.displayName === 'MemoizedFormattedMessage';
