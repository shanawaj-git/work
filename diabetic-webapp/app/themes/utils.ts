
import { diabeticTheme } from './theme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
};

const boxShadowSelector = (level: string): string => BOX_SHADOW_MAPPING[level];

const colorSelector = (color: string): string =>
  diabeticTheme.color[color.toLowerCase()];

export { boxShadowSelector, colorSelector };
