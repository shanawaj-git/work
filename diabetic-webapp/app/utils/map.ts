export const getAddress = (mapOutput): Object => {
  const {
    form: { direction, flatvillaname, flatvillano },
    addressFetched: {
      formatted_address,
      geometry: {
        location: { lat, lng },
      },
    },
  } = mapOutput;
  return {
    flatVillaNumber: flatvillano,
    formattedText: formatted_address,
    buildingName: flatvillaname,
    userNotes: direction,
    latitude: lat,
    longitude: lng,
  };
};
