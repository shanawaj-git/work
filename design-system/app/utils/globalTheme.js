// eslint-disable-next-line import/no-cycle
import { getTheme } from 'themes/getTheme';
import { SETTINGS } from '../config/organizations';

let currentTheme = 'DEFAULT';
const newThemes = {};

const validateTheme = type => {
  if (!SETTINGS[type])
    throw new Error('ERROR : Theme name provided is not defined');
};

class Theme {
  getInstance() {
    return this;
  }

  getTheme() {
    return currentTheme;
  }

  getDynamicThemes() {
    return newThemes;
  }

  setTheme(theme) {
    if (typeof theme === 'string') {
      validateTheme(theme);
      currentTheme = theme;
    } else {
      newThemes[theme.name] = theme;
      currentTheme = theme.name;
    }
  }

  getActiveTheme() {
    return getTheme();
  }
}
export const GlobalTheme = new Theme();
