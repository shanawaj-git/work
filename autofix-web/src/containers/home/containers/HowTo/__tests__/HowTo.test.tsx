import React from "react";
import { HowTo } from "..";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { withWrapper } from "@/utils/IntlProviderWrapper";

const homeActions = [
  {
    desc: "and find out what is wrong with your car or choose a specific service.",
    icon: "http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/option1_a78924043f.svg",
    name: "Run diagnostics",
  },
  {
    desc: "for various services and parts, as well as an estimated time until completion.",
    icon: "http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/icon2_5537eea71f.svg",
    name: "See approximate prices",
  },
  {
    desc: "in one of workshops based on reviews and compare prices.",
    icon: "http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/icon3_c3ace407ba.svg",
    name: "Book a meeting",
  },
];

describe("<HowTo />", () => {
  test("it should mount", () => {
    const tree = render(withWrapper(<HowTo actions={homeActions} />));
    expect(tree.getAllByTestId("howto-section")).toBeTruthy();
  });
  test("it should mount and match the snapshot", () => {
    const tree = render(withWrapper(<HowTo actions={homeActions} />));
    expect(tree.container.innerHTML).toMatchSnapshot();
  });
});
