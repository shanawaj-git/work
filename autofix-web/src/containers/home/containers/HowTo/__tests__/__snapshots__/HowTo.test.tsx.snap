// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowTo /> it should mount and match the snapshot 1`] = `
"<div data-testid=\\"howto-section\\" class=\\"min-h-screen flex flex-col justify-between py-[100px] space-y-8\\"><div class=\\"px-6 \\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 leading-10 text-[34px]  css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> We are here to make 
your car service easy &amp; 
 transparent</span></p><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\" style=\\"border-color: #e25f6d;\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 text-base font-light css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> Enjoy a transparent, easy process with no 
 hidden fees or catches and get affordable 
 car servicing at our nationwide network of 
 independent garages.</span></p></div></div><div class=\\"flex flex-col items-center justify-center w-full px-6 space-y-4\\"><div class=\\"sc-e16d83bc-0 jFceXq w-[327px] md:w-[600px] mx-0\\"><div class=\\"StyledImageCardWithCta__ImageCardWithCtaWrapper-sc-1ntrkoy-0 jcDmhc\\"><img class=\\"image-card-icon\\" src=\\"http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/option1_a78924043f.svg\\" alt=\\"Run diagnostics\\"><div class=\\"image-card-content\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 image-card-title css-11estw0-MuiTypography-root\\">Run diagnostics</h5><span class=\\"MuiTypography-root MuiTypography-caption image-card-subTitle css-1393mlh-MuiTypography-root\\">and find out what is wrong with your car or choose a specific service.</span></div></div></div><div class=\\"sc-e16d83bc-0 jFceXq w-[327px] md:w-[600px] mx-0\\"><div class=\\"StyledImageCardWithCta__ImageCardWithCtaWrapper-sc-1ntrkoy-0 jcDmhc\\"><img class=\\"image-card-icon\\" src=\\"http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/icon2_5537eea71f.svg\\" alt=\\"See approximate prices\\"><div class=\\"image-card-content\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 image-card-title css-11estw0-MuiTypography-root\\">See approximate prices</h5><span class=\\"MuiTypography-root MuiTypography-caption image-card-subTitle css-1393mlh-MuiTypography-root\\">for various services and parts, as well as an estimated time until completion.</span></div></div></div><div class=\\"sc-e16d83bc-0 jFceXq w-[327px] md:w-[600px] mx-0\\"><div class=\\"StyledImageCardWithCta__ImageCardWithCtaWrapper-sc-1ntrkoy-0 jcDmhc\\"><img class=\\"image-card-icon\\" src=\\"http://strapi-dev-autofix.uaenorth.azurecontainer.io:1337/uploads/icon3_c3ace407ba.svg\\" alt=\\"Book a meeting\\"><div class=\\"image-card-content\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 image-card-title css-11estw0-MuiTypography-root\\">Book a meeting</h5><span class=\\"MuiTypography-root MuiTypography-caption image-card-subTitle css-1393mlh-MuiTypography-root\\">in one of workshops based on reviews and compare prices.</span></div></div></div></div></div>"
`;
